### ISSUE TITLE
Is the SciPyInterface really communicating gradients to the torch.nn.Module objects ?

### ISSUE THREAD

**Issue by Joao-L-S-Almeida** [IBM/simulai#129]
It seems that optimizers like L-BFGS-B are not being really effective.
